src/runtime_src/core/pcie/driver/linux/xocl/subdev/p2p.c

```diff
@@ -1087,6 +1087,43 @@ static int p2p_refresh_rbar(struct platform_device *pdev)
 	return 0;
 }
 
+static int p2p_get_bar_paddr(struct platform_device *pdev, ulong bank_addr,
+			     ulong bank_size, ulong *bar_paddr)
+{
+	struct p2p *p2p = platform_get_drvdata(pdev);
+	ulong bank_off = 0;
+	int ret;
+
+	if (p2p->p2p_bar_idx < 0) {
+		p2p_err(p2p, "can not find p2p bar");
+		return -EINVAL;
+	}
+
+	mutex_lock(&p2p->p2p_lock);
+
+	if (bank_addr == ~0UL && p2p->remapper) {
+		/* do not need to reserve bar space if remapper present */
+		ret = 0;
+		goto  failed;
+	}
+
+	bank_off = p2p_bar_map(p2p, bank_addr, bank_size,
+		XOCL_P2P_CHUNK_SIZE);
+	if (bank_off < 0) {
+		ret = -ENOENT;
+		goto failed;
+	}
+
+	p2p_bar_unmap(p2p, bank_off);
```

Comment:
The released map will be reused by next map request. And it looks the same with p2p_remap_resource()? Maybe you may change p2p_remap_resource() and call it? you do not have to release if you will used the address for the entire lifecycle of xclbin.

Reply:
Oh.. let me investigate p2p_remap_resource(). Thanks for the comment.